coral/src/app/features/topics/acl-request/TopicAclRequest.tsx

```diff
@@ -67,24 +67,26 @@ const TopicAclRequest = () => {
     aclType === "PRODUCER"
       ? topicProducerForm.watch("topicname")
       : topicConsumerForm.watch("topicname");
-  useQuery<TopicTeam, Error>({
-    queryKey: ["topicTeam", selectedTopicName, selectedPatternType, aclType],
-    queryFn: () =>
-      getTopicTeam({
-        topicName: selectedTopicName,
-        patternType: selectedPatternType,
-      }),
-    onSuccess: (data) => {
-      if (data === undefined) {
-        throw new Error("Could not fetch team for current Topic");
-      }
-      return aclType === "PRODUCER"
-        ? topicProducerForm.setValue("teamname", data.team)
-        : topicConsumerForm.setValue("teamname", data.team);
-    },
-    enabled: selectedTopicName !== undefined,
-    keepPreviousData: true,
-  });
+  useQuery<TopicTeam, Error>(
+    ["topicTeam", selectedTopicName, selectedPatternType, aclType],
+    {
+      queryFn: () =>
+        getTopicTeam({
+          topicName: selectedTopicName,
+          patternType: selectedPatternType,
+        }),
+      onSuccess: (data) => {
+        if (data === undefined) {
+          throw new Error("Could not fetch team for current Topic");
+        }
+        return aclType === "PRODUCER"
+          ? topicProducerForm.setValue("teamname", data.team as string)
```

Comment:
A note (nothing to do for now as this is the fix branch!) - in this cases I would vote for us to treat the API as "always right" and assume (and code in a way) that team can be undefined, so add a handling for that. That way we make sure that we don't hide potential compiler errors in the future by typecasting - e.g. when for some reason `team` could also be a number (which in this case would not break bc it's the url param ^^ but still should come up as an ts error)

Reply:
Yes, and in this case the definition *is* right. Those properties are undefined when `error` is *not* undefined!

This was a recent change, and the error handling was modified here: https://github.com/aiven/klaw/pull/843/commits/cb1164cded35d19a04bd8d87272d0079ad73cf40

I think it was not enough to satisfy TS thoguh, so I will push this, which removes the need for the coercion

```tsx
      onSuccess: (data) => {
        if (data.error !== undefined || data.team === undefined) {
          throw new Error(data.error);
        }
        return aclType === "PRODUCER"
          ? topicProducerForm.setValue("teamname", data.team)
          : topicConsumerForm.setValue("teamname", data.team);
      },
```

Reply:
[6b72be5](https://github.com/aiven/klaw/pull/843/commits/6b72be53e07135a52617782e7908cc6ac1e685dc)

Reply:
> Yes, and in this case the definition is right. Those properties are undefined when error is not undefined!
💡  

Ok, makes sense. 

Still, this is implicit knowledge that we're using here. Without it, I would translate these lines to: "We're throwing an error if error is undefined but team empty, even though it's ok for team to be empty and it's not an actual error". 

I think at least adding a comment with the info would help. In general I feel (in the future) we should treat the api definition as if it's the only information we have (as others would do) and have a path where team is `undefined` (e.g. just do nothing ^^) instead of throwing an Error. Ugh, I hope I could make clear what I mean XD

Reply:
Added comment here ^^ [9ef23b9](https://github.com/aiven/klaw/pull/843/commits/9ef23b9334fd349f19cd1d646b4e6ed87384542b)